src/main.tf

```diff
@@ -118,6 +118,57 @@ resource "azurerm_linux_function_app" "starter" {
   }
 }
 
+locals {
+  default_frontend_endpoint_name      = "${var.naming_prefix}-${var.environment}-fd-azurefd-net"
+  default_frontend_endpoint           = "${var.naming_prefix}-${var.environment}-fd.azurefd.net"
+}
+
+resource "azurerm_frontdoor" "starter" {
+  name                = "${var.naming_prefix}-${var.environment}-fd"
+  friendly_name       = "${var.naming_prefix}-${var.environment}-fd"
+  resource_group_name = data.azurerm_resource_group.starter.name
+
+  frontend_endpoint {
+    name      = local.default_frontend_endpoint_name
+    host_name = local.default_frontend_endpoint
+  }
+
+  backend_pool {
+    name = "${var.naming_prefix}-${var.environment}-backend-pool"
+    backend {
+      host_header = azurerm_linux_web_app.starter.default_hostname
+      address     = azurerm_linux_web_app.starter.default_hostname
+      http_port   = 80
+      https_port  = 443
+    }
+    load_balancing_name = "load-balancing-${var.naming_prefix}-${var.environment}"
+    health_probe_name   = "health-probe-${var.naming_prefix}-${var.environment}"
+  }
+
+  backend_pool_load_balancing {
+    name = "load-balancing-${var.naming_prefix}-${var.environment}"
+  }
+
+  backend_pool_health_probe {
+    name                = "health-probe-${var.naming_prefix}-${var.environment}"
+    enabled             = false
+    interval_in_seconds = 30
+    probe_method        = "HEAD"
+    protocol            = "Https"
+  }
+
+  routing_rule {
+    name               = "rule-${var.naming_prefix}-${var.environment}-all-paths"
+    accepted_protocols = ["Http", "Https"]
```

Comment:
I don't like this because it allows for an insecure pattern. It allows AFD to accept unencrypted HTTP requests on the frontend and forward them to the backend as HTTPS requests. This would make things "look" secure and encrypted to Azure App Service even though it potentially was **_not_** encrypted between the client and AFD.

I would suggest having a separate AFD rule which returns a redirect response for all HTTP requests to redirect the client to the HTTPS path. (This is what is setup on the Ops repo).

Reply:
Added HTTPS conversion route.